internal/services/cosmos/cosmosdb_account_resource.go

```diff
@@ -463,6 +476,58 @@ func resourceCosmosDbAccount() *pluginsdk.Resource {
 
 			"cors_rule": common.SchemaCorsRule(),
 
+			"restore": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"restore_source": {
+							Type:     pluginsdk.TypeString,
+							Required: true,
+							ForceNew: true,
+							ValidateFunc: validation.All(
+								validation.StringMatch(regexp.MustCompile(`^(.)+/restorableDatabaseAccounts/(.)+$`), "It includes the segment `/restorableDatabaseAccounts/`"),
```

Comment:
can we use the correct resource id validator here?

Reply:
Updated